### PR TITLE
EVG-7499 maintain settings for distro UI

### PR DESCRIPTION
**Pull request by ablack12**
Distro UI not loading because updating distro settings in erroneously removing the legacy provider settings structure, which the current distro UI is still using (to be changed in EVG-7233).

### CODE REVIEW

File: cloud/docker.go [hunk 1]
```diff
@@ -45,18 +45,18 @@ func (settings *dockerSettings) Validate() error {
...
-	} else if len(d.ProviderSettingsList) != 0 {
+	if len(d.ProviderSettingsList) != 0 {
 		bytes, err := d.ProviderSettingsList[0].MarshalBSON()
 		if err != nil {
 			return errors.Wrap(err, "error marshalling provider setting into bson")
 		}
 		if err := bson.Unmarshal(bytes, s); err != nil {
 			return errors.Wrap(err, "error unmarshalling bson into provider settings")
 		}
+	} else if d.ProviderSettings != nil {
```
Comment (john-m-liu) [inline cloud/docker.go]: Does it make sense to take out the else in these? Like are there situations where both will be populated?

File: service/distros.go [hunk 2]
```diff
@@ -122,6 +122,13 @@ func (uis *UIServer) modifyDistro(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// ensure docker password wasn't auto-filled from form
+	if newDistro.Provider != evergreen.ProviderNameDocker && newDistro.Provider != evergreen.ProviderNameDockerMock {
```
Comment (john-m-liu) [inline service/distros.go]: Is it possible to make this more generalized so that we delete all settings that aren't relevant to the type?

Comment (ablack12) [reply service/distros.go]: I thought about that, but not without roundtripping into the right settings object which feels like overkill--I think this problem will go away with the distro ticket mentioned above anyway (or I could stop using that password form field thing) so this is more a temporary fix.

File: cloud/docker.go (hunk 1 above)
Comment (ablack12) [reply cloud/docker.go]: Yeah there are, this whole thing is wonky and I don't like it

Comment (john-m-liu) [reply cloud/docker.go]: I think that means we don't want the "else" right? Like we want both if statements to be checked?

Comment (ablack12) [reply cloud/docker.go]: No, it's like "if the new structure is available, populate using that, else populating using the legacy"

**Pull request merged**